### PR TITLE
Multiple clients

### PR DESCRIPTION
**Pull request by karl-thomas**

### CODE REVIEW

**Comment by lpenzey** [comment]
Looking good! 

Only major thing to add would be a `readme`.

File: src/test/java/com/karl/ServerIT.java [hunk 1]
```diff
@@ -19,13 +19,13 @@
   class when_server_is_started {
     private Client client;
 
-    @BeforeAll
```
Comment (lpenzey) [inline src/test/java/com/karl/ServerIT.java:22]: `ServerIT` == `ServerIntegrationTests`?

File: src/test/java/com/karl/ServerIT.java [hunk 2]
```diff
@@ -48,5 +48,21 @@ public void echoMultipleMessageFromSameClient() throws IOException {
...
+    @Test
+    @DisplayName("server can respond to multiple messages, one client at a time")
+    public void echoAlternatingClients() throws IOException {
+      String msg1 = "Howdy";
+      String resp1 = client.sendMessage(msg1);
+      client.close();
+
+      client = new Client();
+      client.connectTo(Globals.HOST, Globals.PORT);
+      String msg2 = "Pard'ner";
```
Comment (lpenzey) [inline src/test/java/com/karl/ServerIT.java:61]: 🤠

File: src/test/java/com/karl/ServerTest.java [hunk 3]
```diff
@@ -4,38 +4,62 @@
...
   public void serverCanConnectToSocket() throws IOException {
-    MockSocketWrapper socket = new MockSocketWrapper();
-    MockServerSocket serverSocket = new MockServerSocket();
-    serverSocket.socket = socket;
+    assertEquals(server.accept(), mocket);
+  }
 
-    Server server = new Server(serverSocket);
+  @Test
+  @DisplayName("server will not write anything if the client didn't send anthing")
```
Comment (lpenzey) [inline src/test/java/com/karl/ServerTest.java:39]: 🐜

File: src/test/java/com/karl/ServerTest.java [hunk 4]
```diff
@@ -4,38 +4,62 @@
...
 import org.junit.jupiter.api.Test;
 
 @Tag("unit")
 public class ServerTest {
+  private MockSocketWrapper mocket;
+  private MockServerSocket serverSocket;
+  private Server server;
+
+  @BeforeEach
+  public void setupEach() throws IOException {
```
Comment (lpenzey) [inline src/test/java/com/karl/ServerTest.java:20]: digging this `BeforeEach` server setup 🎉

**Review by lpenzey — APPROVED** [review]

File: src/test/java/com/karl/ServerIT.java (hunk 1 above)
Comment (karl-thomas) [reply src/test/java/com/karl/ServerIT.java:22]: yes, its the default convention for the testrunner being used, and i was too lazy to have it search for a new, more descriptive file blob.

Comment (lpenzey) [reply src/test/java/com/karl/ServerIT.java:22]: Yeah no worries, `ServerIT` works for me just wanted to make sure I understood.

File: src/test/java/com/karl/ServerTest.java (hunk 3 above)
Comment (lpenzey) [reply src/test/java/com/karl/ServerTest.java:39]: (this was in reference to a typo in `anthing` sorry if that wasn't clear)

**Pull request merged**